lib/maths/time_series/CTimeSeriesDecompositionDetail.cc

```diff
@@ -1177,6 +1190,10 @@ void CTimeSeriesDecompositionDetail::CSeasonalityTest::handle(const SAddValue& m
 
     switch (m_Machine.state()) {
     case PT_TEST:
+        // break immediately if in hard_limit state
```

Comment:
I would explain this instead, i.e. something like "The seasonality test memory can  increase as we add new values so we stop updating it in hard limit."

Reply:
comment added as you suggested.